sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceRDD.scala

```diff
@@ -61,6 +61,18 @@ class DataSourceRDD(
       private var currentIter: Option[Iterator[Object]] = None
       private var currentIndex: Int = 0
 
+      private val partitionMetricCallback = if (columnarReads) {
+        new PartitionMetricCallback[ColumnarBatch](customMetrics)
+      } else {
+        new PartitionMetricCallback[InternalRow](customMetrics)
+      }
```

Comment:
nit: We dont really need the if/else here ... the bound gets ignored.

Reply:
Removed. Thanks.